File: progetto-rn/styles/global.js
import { StyleSheet } from 'react-native';
import colors from './colors';

const fontSizes = {
	verySmall: 10,
	small: 14,
	normal: 16,
	subtitle: 20,
	title: 24,
}

export const fonts = {
	regular: 'UberMoveText-Regular',
	medium: 'UberMoveText-Medium',
	bold: 'UberMoveText-Bold',
	logo: 'Geologica-Roman-Medium',
};

/**
 * @typedef {Object} ColorStyle
 * @property {string} [color]
 * @property {string} [backgroundColor]
 */

/**
 * @typedef {Object<string?, ColorStyle>} ColorStyles
 */

/**
 * @returns {ColorStyles}
 */
function colorStyles() {
	let styles = {}
	for (const [key, value] of Object.entries(colors)) {
		const capitalizedKey = key.charAt(0).toUpperCase() + key.slice(1);
		styles["text" + capitalizedKey] = {
			color: value,
		};
		styles["bg" + capitalizedKey] = {
			backgroundColor: value,
		};
	}
	return styles
}

/**
 * @typedef {Object} TypographyStyle
 * @property {number} fontSize
 * @property {string} fontFamily
 */

/**
 * @typedef {Object<string?, TypographyStyle>} TypographyStyles
 */

/**
 * @returns {TypographyStyles}
 */
function typographyStyles() {
	let styles = {}
	for (const [fontSize, fontSizeValue] of Object.entries(fontSizes)) {
		const capitalizedFontSize = fontSize.charAt(0).toUpperCase() + fontSize.slice(1);
		for (const [fontWeight, fontWeightValue] of Object.entries(fonts)) {
			const capitalizedFontWeight = fontWeight.charAt(0).toUpperCase() + fontWeight.slice(1);
			styles["text" + capitalizedFontSize + capitalizedFontWeight] = {
				fontSize: fontSizeValue,
				fontFamily: fontWeightValue,
			};
		}
	}
	return styles
}

export const globalStyles = StyleSheet.create({
	...colorStyles(),
	...typographyStyles(),

	container: {
		backgroundColor: colors.white,
		paddingVertical: 8,
		flex: 1,
	},

	insetContainer: {
		backgroundColor: colors.white,
		paddingHorizontal: 20,
	},

	flexBetween: {
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'space-between'
	},

	flexCenter: {
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'center',
	}

});

File: progetto-kt/app/src/main/java/com/example/progetto_kt/view/styles/Global.kt
package com.example.progetto_kt.view.styles

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.progetto_kt.R

object Global {

    data object Fonts {
        val Regular = FontFamily(Font(R.font.ubermovetext_regular))
        val Medium = FontFamily(Font(R.font.ubermovetext_medium))
        val Bold = FontFamily(Font(R.font.ubermovetext_bold))
        val Logo = FontFamily(Font(R.font.geologica_medium))
    }

    data object FontSizes {
        val VerySmall = 10.sp
        val Small = 14.sp
        val Normal = 16.sp
        val Subtitle = 20.sp
        val Title = 24.sp
    }

    val Container = Modifier
        .background(Colors.WHITE)
        .padding(vertical = 8.dp)
        .fillMaxWidth()

    val InsetContainer = Modifier
        .background(Colors.WHITE)
        .padding(horizontal = 20.dp)
        .fillMaxWidth()


}